validator/client/validator.go

```diff
@@ -1050,7 +1050,8 @@ func (v *validator) filterAndCacheActiveKeys(ctx context.Context, pubkeys [][fie
 			}
 			v.pubkeyToValidatorIndex[k] = i
 		}
-		statusRequestKeys = append(statusRequestKeys, k[:])
+		copiedk := k
+		statusRequestKeys = append(statusRequestKeys, copiedk[:])
```

Comment:
There is a static analysis that catches this, i wonder why it didnt do it here? It's called loopclosure.